sycl/doc/extensions/proposed/sycl_ext_oneapi_usm_properties.asciidoc

```diff
@@ -0,0 +1,185 @@
+below.  Applications can test for the existence of this macro to determine if
+the implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's features the implementation
+supports.
+
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|Initial version of this extension.
+|===
+
+
+== Contributors
+
+Abhishek Tiwari, Intel +
+Aditi Kumaraswamy, Intel +
+Gregory Lueck, Intel +
+Jason Sewall, Intel +
+Jessica Davies, Intel +
+Joe Garvey, Intel +
+Mike Kinsner, Intel +
+Sherry Yuan, Intel
+
+
+=== Examples
+
+The following is a synopsis of the new USM malloc API.
+
+[source,c++]
+----
+
+// A property list containing a compile-time property and a runtime property
+sycl::ext::oneapi::experimental::properties properties{some_compile_time_prop<1>, some_runtime_prop(1)};
+
+// The compile time property is passed to template arguments of annotated_ptr
+auto data = sycl::ext::oneapi::experimental::malloc_device(N, q, properties);
+// data is of type annotated_ptr<void*, property_list_t<some_compile_time_prop<1>>>
+
+auto data2 = sycl::ext::oneapi::experimental::malloc_device<int>(N, q, properties);
+// data2 is of type annotated_ptr<int*, property_list_t<some_compile_time_prop<1>>>
+
+sycl::queue q;
+q.parallel_for(range<1>(N), [=] (id<1> i){
+  data2[i] *= 2;
+}).wait();
+----
+
+
+=== New USM allocation APIs
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+// All function defined below are available only when is_property_list<PropertyListT>::value is true
+
+template <typename T = void, typename PropertyListT>
+annotated_ptr<T, PropertyListT> malloc_device(
+    size_t Count, const queue &Q, PropertyListT &PropList) {
+  ...
+}
+
+template <typename T = void, typename PropertyListT>
+annotated_ptr<T, PropertyListT> malloc_device(
+    size_t Count, const device &Dev, const context &Ctxt, PropertyListT &PropList) {
+  ...
+}
+
+template <typename T = void, typename PropertyListT>
+annotated_ptr<T, PropertyListT> malloc_shared(
+    size_t Count, const queue &Q, PropertyListT &PropList) {
+  ...
+}
+
+template <typename T = void, typename PropertyListT>
+annotated_ptr<T, PropertyListT> malloc_shared(
+    size_t Count, const device &Dev, const context &Ctxt, PropertyListT &PropList) {
+  ...
+}
+
+template <typename T = void, typename PropertyListT>
+annotated_ptr<T, PropertyListT> malloc_host(
+    size_t Count, const context &Ctxt, PropertyListT &PropList) {
+  ...
+}
+
+template <typename T = void, typename PropertyListT>
+annotated_ptr<T, PropertyListT> malloc_host(
+    size_t Count, const queue &Q, PropertyListT &PropList) {
+  ...
+}
+
+} // namespace sycl::ext::oneapi::experimental
+----
+
+The same setup is applied to other overloads of malloc APIs.
+
+The returned `annotated_ptr` contains the USM pointers and the passed in compile-time properties to enable additional compiler optimizations.
```

Comment:
This seems to imply that the annotated_ptr can only contain those properties passed in to the properties argument of the malloc call.  I can think of one situation where we don't want to be that restrictive: if we add a property to encode if an allocation is device vs. host vs. shared we'd like to be able to add that to the annotated_ptr based on which malloc call was made even if the user didn't put the associated property in the properties argument.